src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1432,9 +1444,37 @@ by { ext p, simp [←coe_finset_walk_length_eq] }
 power of the adjacency matrix. -/
 lemma card_set_walk_length_eq (u v : V) (n : ℕ) :
   fintype.card {p : G.walk u v | p.length = n} = (G.finset_walk_length n u v).card :=
-fintype.card_of_subtype (G.finset_walk_length n u v) $ λ p,
+fintype.card_of_finset (G.finset_walk_length n u v) $ λ p,
 by rw [←finset.mem_coe, coe_finset_walk_length_eq]
 
+instance fintype_set_path_length (u v : V) (n : ℕ) :
+  fintype {p : G.walk u v | p.is_path ∧ p.length = n} :=
```

Comment:
IIGTSC (if I get the sectioning correctly), you're proving this for `fintype V`. Can you instead prove it for `Π v, fintype (G.neighbor_set v)`?

Reply:
Yes, I certainly can, thanks. I've generalized things to locally finite graphs as far as possible, though you need finiteness for the new instances.